### ISSUE TITLE
[FEATURE] Alertmanager supports main stream receivers

### ISSUE THREAD

**Issue by w13915984028** [harvester/harvester#2521]
**Is your feature request related to a problem? Please describe.**

For better customized integration of alertmanager, main stream receivers should be supported in Harvester Alertmanager.

**Describe the solution you'd like**

For better customized integration of alertmanager, main stream receivers should be supported in Harvester Alertmanager.

Harvester dashboard supports add/change/remove of alertmanager receiver.  Harvester dashboard supports multi receivers in parallel.

https://github.com/harvester/harvester/issues/2643

**Describe alternatives you've considered**

**Additional context**

The detailed description about the `AlertmanagerConfig` and how to config it, is added in HEP:
HEP PR:
HEP: Harvester supports alertmanager #2545
https://github.com/harvester/harvester/pull/2545

**Comment by w13915984028** [comment]
The UI issue is tracked in 

https://github.com/harvester/harvester/issues/2643

**Comment by w13915984028** [comment]
Known limitation:

The `AlertmanagerConfig` is enforced by `namespace`. But, not all `alerts` having `namespace`. Meanwhile, create same `receiver` in each `namespace` is also tedious.

The is a workaround as per below link. It allows to add a gloabl config to `Alertmanager`, which in turn removes `namespace` check in `AlertmanagerConfig`.

Unfortunately,  the `alertmanagerConfiguration` is supported in Harvester used `rancher-monitoring`. 

https://github.com/prometheus-operator/prometheus-operator/issues/3737#issuecomment-1220906714

**Comment by w13915984028** [comment]
For TEST:

Pleaes first validate the `webhook` type receiver is well supported, it is raised by customer.

Reference:
https://github.com/w13915984028/harvester/blob/fix2517/enhancements/20220720-alertmanager.md#story-3

Before UI is ready, the related config could be applied via `kubectl`.
https://github.com/w13915984028/harvester/blob/fix2517/enhancements/20220720-alertmanager.md#adddelete-receiver-via-crd-alertmanagerconfig

After UI is ready, https://github.com/harvester/harvester/issues/2643, the related config is available in dashboard.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-4c2242cd-head` (09/21). Close this issue.

### Result
Now we can add `webhook` type receiver from the `Altermanager Config` page for the given namespace
![image](https://user-images.githubusercontent.com/29251855/191467324-220f4e30-53fe-4893-963b-439b5dd8446b.png)

![image](https://user-images.githubusercontent.com/29251855/191467902-9a38f735-038a-40c6-94e5-462e48aa1bcf.png)

* Please be noted, currently we can't create the receiver to collect all namespace alert which will be tracked in
https://github.com/harvester/harvester/issues/2760

* Can applied the changes to `alertmanagerconfig` pod in Harveser
```
harv0921:~ # kubectl get alertmanagerconfig -A -o yaml
apiVersion: v1
items:
- apiVersion: monitoring.coreos.com/v1alpha1
  kind: AlertmanagerConfig
  metadata:
    creationTimestamp: "2022-09-21T08:16:21Z"
    generation: 1
    name: alert-receiver
    namespace: default
    resourceVersion: "14566"
    uid: 8f9c30a2-42c1-4831-92b5-8abf71134333
  spec:
    receivers: []
    route:
      groupBy: []
      groupInterval: 5m
      groupWait: 30s
      matchers: []
      repeatInterval: 4h
- apiVersion: monitoring.coreos.com/v1alpha1
  kind: AlertmanagerConfig
  metadata:
    creationTimestamp: "2022-09-21T08:48:38Z"
    generation: 2
    name: alertmanager
    namespace: default
    resourceVersion: "39171"
    uid: 32f813c3-24fd-4faf-a270-1e7723a5dfdf
  spec:
    receivers:
    - name: webhook
      webhookConfigs:
      - httpConfig:
          tlsConfig: {}
        sendResolved: false
        url: http://192.168.122.230:8094
    route:
      groupBy: []
      groupInterval: 5m
      groupWait: 30s
      matchers: []
      repeatInterval: 4h
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

```

### Test Information
* Test Environment: 1 node Harvester on local kvm machine
* Harvester version: `master-4c2242cd-head` (09/21)

### Verify Steps
1. Prepare another VM or machine have the same subnet with the Harvester 
2. Prepare a webhook server on the VM, reference to https://github.com/w13915984028/harvester-develop-summary/blob/main/test-log-event-audit-with-webhook-server.md
3. You may need to install python3 web package, refer to https://webpy.org/install
4. Run `export PORT=8094` on the webhook server VM 
5. Launch the webhook server `python3 simple-webhook-server.py` 
```
davidtclin@ubuntu-clean:~$ python3 simple-webhook-server.py
usage: export PORT=1234 to set http server port number as 1234
start a simple webhook server, PORT 8094 @ 2022-09-21 16:39:58.706792 

http://0.0.0.0:8094/
```
6. Access Harvester `Alertmanager Configs` page in Monitoring 
7. Create an Alertmanager config 
![image](https://user-images.githubusercontent.com/29251855/191452211-9892187b-4af2-4f73-8d34-19fedcd830b8.png)
8. Access the created Alertmanager config 
9. Click Add receiver 
![image](https://user-images.githubusercontent.com/29251855/191452302-929fe806-6485-4837-be84-f2814c28d4ac.png)
10. Select webhook page 
11. Click Add webhook
![image](https://user-images.githubusercontent.com/29251855/191452393-de222b27-9d0a-4dc5-bed0-e9c449b95766.png)
12. Add `http://server_ip:8094 to the URL 
![image](https://user-images.githubusercontent.com/29251855/191458438-98d3e36d-27af-4f26-aaf4-c910036ed124.png)
13. Access Harvester node 
14. Run `kubectl get alertmanagerconfig -A -o yaml`
15. Check the webhook server have applied to the `Altertmanagerconfig` pod 
16. Check the webhook receiver can be created 
![image](https://user-images.githubusercontent.com/29251855/191464505-d1a2d718-a742-4072-853a-a4c221abe97a.png)

### Additional Context 
Refer to HEP document for more details 
https://github.com/w13915984028/harvester/blob/fix2517/enhancements/20220720-alertmanager.md
